tests/playwright/sanity/container-grid.test.ts

```diff
@@ -544,6 +544,47 @@ test.describe( 'Container Grid tests @container', () => {
 			await expect( editor.getPreviewFrame().locator( '.elementor-first-add' ) ).toHaveCount( 1 );
 		} );
 	} );
+
+	test( 'Test Empty View should not distorse preview', async ( { page }, testInfo ) => {
+		const wpAdmin = new WpAdminPage( page, testInfo ),
+			editor = await wpAdmin.useElementorCleanPost();
+
+		let containerId;
+		let gridRowsControl;
+
+		await test.step( 'Arrange', async () => {
+			await editor.closeNavigatorIfOpen();
+			containerId = await editor.addElement( { elType: 'container' }, 'document' );
+			await editor.setSelectControlValue( 'container_type', 'grid' );
+
+			gridRowsControl = page.locator( '.elementor-control-grid_rows_grid' );
+			const columnsRowControl = page.locator( '.elementor-control-grid_columns_grid' );
+
+			await columnsRowControl.locator( '.elementor-slider-input input' ).fill( '3' );
+			await gridRowsControl.locator( '.elementor-slider-input input' ).fill( '2' );
+
+			for ( let i = 0; i < 6; i++ ) {
+				await editor.addWidget( 'heading', containerId );
+			}
+		} );
+
+		await test.step( 'Empty item should not be presented', async () => {
+			await expect( editor.getPreviewFrame().locator( '.elementor-empty-view' ) ).toBeHidden();
+		} );
+
+		await test.step( 'Add one more item to see empty item', async () => {
+			const latestAddedWidgetId = await editor.addWidget( 'heading', containerId );
+			await expect( editor.getPreviewFrame().locator( '.elementor-empty-view' ) ).toBeVisible();
+
+			await editor.removeElement( latestAddedWidgetId );
+		} );
+
+		await test.step( 'Increase number of rows to see empty item', async () => {
+			await editor.selectElement( containerId );
+			await gridRowsControl.locator( '.elementor-slider-input input' ).fill( '3' );
+			await expect( editor.getPreviewFrame().locator( '.elementor-empty-view' ) ).toBeVisible();
+		} );
+	} );
```

Comment:
I would add a test that checks the device mode change.
E,g. On the desktop, the grid is 4*3 and contains 12 items. on the tablet, the grid is 3*3 so in the tablet we expect to see the empty view.